src/cache.ts

```diff
@@ -73,8 +77,17 @@ export class ListWatch<T extends KubernetesObject> implements ObjectCache<T>, In
     }
 
     private async doneHandler(err: any) {
+        if (this.request != null) {
+            this.request.destroy();
+            this.request = null;
+        }
         if (err) {
-            this.callbackCache[ERROR].forEach((elt: ObjectCallback<T>) => elt(err));
+            // On an error, the done handler is called twice with the same error, see details in
+            // watch.ts
+            if (!this.stopped) {
+                this.stopped = true;
+                this.callbackCache[ERROR].forEach((elt: ObjectCallback<T>) => elt(err));
```

Comment:
I don't see that we would set `this.stopped` to false again in the start() method. If I understand correctly it prevents the cache from being reused. And if we set `stopped` to `false` in `start()` method then it would create a race condition when `doneHandler` is called twice and `stopped` is set to false between those two calls. To me, that does not look like a good way to go. Fixing the watcher to call doneCallback just once sounds like a better option.